wall_e/extensions/sfu.py

```diff
@@ -502,6 +506,104 @@ async def outline(self, ctx, *course):
         if e_obj is not False:
             await ctx.send(embed=e_obj, reference=ctx.message)
 
+    @app_commands.command(name="courses", description="Gets all offered courses")
+    @app_commands.describe(department="Specify the department to search. Examples: STAT, PHYS, MATH")
+    @app_commands.describe(level="Specifies the level of courses to filter for. Examples: 100, 200, 400")
+    @app_commands.describe(term="Specify the specific semester to search for. Examples: Spring, Summer, Fall")
+    @app_commands.describe(year="Specify the specific year to search for. Examples: 2020, 2024, 2025")
+    async def courses(self, interaction: discord.Interaction, department: str = "", level: int = 0,
```

Comment:
Is it possible to use an optional type instead of having level = 0 refer to "any level"?
```suggestion
    async def courses(self, interaction: discord.Interaction, department: str = "", level: Optional[int] = 0,
```

If that's too cumbersome for discord.py, let's at least use a constant `ANY_LEVEL = 0` (probably defined just above this function?) in the rest of this function, so it's clear what 0 refers to.

Reply:
i never used `Optional` before but why exactly do we use it and how does it make clear what 0 refers to?

Reply:
Optional is a type that's either `int` or `None`. Depending on the python version, it might be the hint `None | int` (newer than 3.10 iirc). If `level is None`, then it's more clear that it's not been defined, where `0` being an integer is still a number.

I also notice I accidentally forgot to include a default value of `None` for the suggestion:
```suggestion
    async def courses(self, interaction: discord.Interaction, department: str = "", level: Optional[int] = None,
```

Like before, it's up to you which approach you prefer!

Reply:
https://github.com/CSSS/wall_e/pull/666/commits/5f54299272d64864628b480c63f3459a4236650e